serde-redis-stream-derive/src/lib.rs

```diff
@@ -75,10 +75,51 @@ pub fn redis_stream_serialize(input: TokenStream) -> TokenStream {
         .map(|field| {
             let f_ident = field.ident.expect("struct fields must have ident");
             let f_lit = f_ident.to_string();
-            let f_type = match field.ty{
-                syn::Type::Path(not_type) => not_type.path.get_ident().map(|value| value.to_owned()).unwrap(),
+            let (f_type,is_option) = match field.ty{
+                syn::Type::Path(not_type) => {                   
+                    let segments_str = &not_type.path.segments
+                    .iter()
+                    .map(|segment| segment.ident.to_string())
+                    .collect::<Vec<_>>()
+                    .join(":");                 
+
+                    let is_option = ["Option", "std:option:Option", "core:option:Option"]
```

Comment:
Yes, fully qualified path names use "::" indeed. But, at this point, we are not comparing the fully qualified name **itself**. 

What's being compared here is what the code built above from the path segments. In other words, the code above is building the full path from a segment list in an arbitrary way. Therefore, we could have used any marker, including ':' (line 84). 

Does it make more sense to use "::"? 
I don't know. That makes sense.

Is that a problem? 
No, because changing the marker above ( ':' -> "::") would only change the array of possible paths for Option  (line 86) ( "Option", "std:option:Option", "core:option:Option" -> "Option", "std::option::Option", "core::option::Option") . 

_Just to be clear:_ 
**Is the code on line 86 a wrong fully qualified path name?**
No, it is not a fully qualified name. It is just an arbitrary way of building the type of the full path.

**Does the code above work**
Yes.